### PR TITLE
Add support for securityContext for various pods

### LINKED ISSUE #466: securityContext not respected in console deployments (and therefore pods)
## Expected Behavior
Setting the security context should apply to all pods created by the operator. Alternatively, there should be a configuration option for separate pods to set the securityContext.

## Current Behavior
Setting the securityContext only applies to the minio StatefulSet, and not the console deployment.

## Possible Solution
Apply the securityContext to other minio components as well.

## Steps to Reproduce (for bugs)
1. Create a minio tenant with a `securityContext` and console
2. Observe it is not propagated to the console deployment/pod

## Context
I am attempting to use the minio operator on a cluster where running containers with the root user is prohibited (both as an official policy and technically via a PodSecurityPolicy). The console pod gets stuck at `CreateContainerConfigError` with the error:

```
Error: container has runAsNonRoot and image will run as root (pod: "example-console-5fcdc814d-2zm7h_namespace(8061bfa4-f11e-433d-9f99-15794e36c45d)", container: console)
```

## Regression
Not a regression.

## Your Environment
N/A (applies to all environments)

### PR DESCRIPTION
**Pull request by nitisht**
This PR address support for specific securityContext for all
the resources created by Operator

Fixes #466

### CODE REVIEW

File: pkg/apis/minio.min.io/v1/conversion.go [hunk 1]
```diff
@@ -140,7 +143,7 @@ func (dst *Tenant) ConvertFrom(srcRaw conversion.Hub) error { //nolint
 	dst.Spec.RequestAutoCert = src.Spec.RequestAutoCert
 	dst.Spec.S3 = src.Spec.S3
 	dst.Spec.CertConfig = src.Spec.CertConfig
-	dst.Spec.SecurityContext = src.Spec.SecurityContext
+	dst.Spec.SecurityContext = src.Spec.Pools[0].SecurityContext
```
Comment (Alevsk) [inline pkg/apis/minio.min.io/v1/conversion.go:146]: we should check here `len(src.Spec.Pools) > 0`

File: pkg/resources/statefulsets/minio-statefulset.go [hunk 2]
```diff
@@ -245,11 +245,11 @@ func minioPoolTolerations(z *miniov2.Pool) []corev1.Toleration {
...
 }
 
-// Builds the security context for a Tenant
-func minioSecurityContext(t *miniov2.Tenant) *corev1.PodSecurityContext {
+// Builds the security context for a Pool
+func minioSecurityContext(pool *miniov2.Pool) *corev1.PodSecurityContext {
 	var securityContext = corev1.PodSecurityContext{}
-	if t.Spec.SecurityContext != nil {
-		securityContext = *t.Spec.SecurityContext
+	if pool.SecurityContext != nil {
```
Comment (Alevsk) [inline pkg/resources/statefulsets/minio-statefulset.go:251]:
```suggestion
	if pool != nil && pool.SecurityContext != nil {
```

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: pkg/apis/minio.min.io/v1/conversion.go [hunk 3]
```diff
@@ -84,6 +83,10 @@ func (src *Tenant) ConvertTo(dstRaw conversion.Hub) error {
 	dst.Spec.ImagePullPolicy = src.Spec.ImagePullPolicy
 	dst.Spec.SideCars = src.Spec.SideCars
 	dst.Spec.ExposeServices = src.Spec.ExposeServices
+	// Apply the securityContext to all the Pools
+	for _, p := range dst.Spec.Pools {
```
Comment (dvaldivia) [inline pkg/apis/minio.min.io/v1/conversion.go:87]: is the security context per pool on v1?

File: examples/tenant.yaml [hunk 4]
```diff
@@ -113,6 +113,12 @@ spec:
       #     cpu: 500m
       #      memory: 16Gi
 
+      ## Configure security context 
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+        runAsNonRoot: true
```
Comment (kerneltime) [inline examples/tenant.yaml:120]: Why support root as true? We should never need to run console or minio as root.

File: pkg/apis/minio.min.io/v1/conversion.go (hunk 3 above)
Comment (nitisht) [reply pkg/apis/minio.min.io/v1/conversion.go:87]: No, it is per pool on V2. So while conversion I am setting the SecurityContext for all the pools

File: examples/tenant.yaml (hunk 4 above)
Comment (nitisht) [reply examples/tenant.yaml:120]: This is not a Operator / Tenant feature. I am just giving an example to disable running as root user in the `securityContext`.

**Review by dvaldivia — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**